File: 프로그래머스/전화번호목록.java
import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        Arrays.sort(phone_book);
        
        for(int i = 0 ; i < phone_book.length - 1; i++){
            int length = phone_book[i].length();
            
            for(int j = i + 1; j < phone_book.length; j++){
                if(phone_book[i].startsWith(phone_book[j]) || phone_book[j].startsWith(phone_book[i])){
                    return false;
                }
            }
        }
        
        
        return answer;
    }
}

File: 프로그래머스/전화번호목록.py
import copy

def solution(phone_book):
    phone_book = [int(phone_book[i]) for i in range(len(phone_book))]
    phone_book.sort()
    
    for num in phone_book:
        temp_book = copy.deepcopy(phone_book)
        temp_book.remove(num)
        
        temp_book = [str(temp_book[i])[:len(str(num))] for i in range(len(temp_book))]
        print(temp_book)

        if str(num) in temp_book:
            return False
    return True